src/main/java/net/schmizz/sshj/sftp/SFTPEngine.java

```diff
@@ -81,7 +81,18 @@ public String canonicalize(String path)
 
     public SFTPEngine init()
             throws IOException {
-        transmit(new SFTPPacket<Request>(PacketType.INIT).putUInt32(MAX_SUPPORTED_VERSION));
+        return init(MAX_SUPPORTED_VERSION);
+    }
+
+    public SFTPEngine init(int requestedVersion)
```

Comment:
Make this method package protected, so the test can call it, but it not becomes part of the contract.

Reply:
No problem :)
To make this change I had to create a new test class in the net.schmizz.sshj.sftp package. 
Hope this is ok.